modules/common/src/test/scala/com.snowplowanalytics.snowplow.enrich.common/utils/JsonPathSpec.scala

```diff
@@ -88,4 +93,23 @@ class JsonPathSpec extends Specification {
 
   def e6 =
     JsonPath.query("$.store.book[2]", Json.fromString("somestring")) must beRight(List())
+
+  def e7 =
+    JsonPath.anyToJson(null) must be(Json.Null)
+
+  def e8 = {
+    JsonPath.anyToJson(5) must beEqualTo(Json.fromLong(5))
+    JsonPath.anyToJson(-1) must beEqualTo(Json.fromLong(-1))
```

Comment:
Am I right that this test would fail with original mapper?

Reply:
I changed the value to a long in my last commit but it doesn't fail using original mapper, since equality check doesn't include type check but value check only. However ValidatorSpec added in last commit fails using original mapper.